Orange/widgets/visualize/owpythagoreanforest.py

```diff
@@ -218,36 +203,55 @@ def _clear_depth_slider(self):
         self.ui_depth_slider.parent().setEnabled(False)
         self.ui_depth_slider.setMaximum(0)
 
-    # HELPFUL METHODS
     def _get_max_depth(self):
         return max([tree.tree_adapter.max_depth for tree in self.ptrees])
 
     def _get_forest_adapter(self, model):
         return SklRandomForestAdapter(model)
 
-    def _draw_trees(self):
+    @contextmanager
+    def disable_ui(self):
+        """Temporarly disable the UI while trees may be redrawn."""
         self.ui_size_calc_combo.setEnabled(False)
+        self.ui_depth_slider.setEnabled(False)
+        self.ui_target_class_combo.setEnabled(False)
+        self.ui_zoom_slider.setEnabled(False)
+        yield
+        self.ui_size_calc_combo.setEnabled(True)
+        self.ui_depth_slider.setEnabled(True)
+        self.ui_target_class_combo.setEnabled(True)
+        self.ui_zoom_slider.setEnabled(True)
```

Comment:
Same here: re-enabling should be in `finally` (and everything before that in `try`).